security-framework/src/trust_settings.rs

```diff
@@ -273,4 +289,23 @@ mod test {
                    .message(),
                    Some("The specified item could not be found in the keychain.".into()));
     }
+
+    #[test]
+    fn test_set_trust_to_cert(){
+        use crate::certificate::SecCertificate;
+        use crate::os::macos::keychain::SecKeychain;
+        use std::{fs, path::Path};
+        
+        let der = fs::read("./test/server.der").unwrap();
+        let cert = SecCertificate::from_der(&der).unwrap();
+
+        let kch = SecKeychain::open(&Path::new("/Library/Keychains/System.keychain")).unwrap();
```

Comment:
This is going to require elevated privileges. Can you make a test that works without extra permissions?